### ISSUE TITLE
Support for Golang NaCl ciphertext output format

### ISSUE THREAD

**Issue by Ajibola** [google/tink#62]
Hi, 

It seems that the java library implements NaCl via NaClCrypto but it is not interoperable with NaCl in Golang, mainly because the chosen output format for cipher text in this library varies.

It would be a great help if interop was considered for this library since this is one of the major issues developers face while implementing crypto across server and client, especially for Android.

Looking forward to the Golang source being updated to mirror the java source. 

Thanks for the good work

**Comment by thaidn** [comment]
Hi Ajibola,

Thanks for filing this request. Could you please point me to the NaCl impl. in Golang? Is it compatible with the original C/C++ NaCl?

**Comment by thaidn** [comment]
Note that our NaClCrypto is @Alpha, and is going to be removed with a different interface.

**Comment by thaidn** [comment]
I found the implementation: https://github.com/golang/crypto/blob/master/nacl/box/box.go.

In NaClBoxCrypto we implemented NaCl's box using the HybridEncrypt and HybridDecrypt interface, but this is actually wrong because NaCl's box is authenticated public key encryption while our hybrid encryption doesn't provide authenticity (but only secrecy).

Thus in order to implement NaClBoxCrypto, first we'll need to add a new interface. We're working on that and I'll update this bug once it's done.

We also provide secret_box. I believe the only interoperability issue is we're using ciphertext || tag, but NaCl is using tag || ciphertext. This could be easily fixed.
